src/components/results/list/mixins/FileListMixin.js

```diff
@@ -0,0 +1,83 @@
+import ListBase from '@/components/results/list/ListBase';
+import store from '@/store';
+import getResourceURL from '@/helpers/resourceURL';
+import { pageSize } from '@/helpers/ApiHelper';
+
+/**
+ * this mixin makes file lists load their results and allows navigation
+ */
+export default {
+  components: {
+    ListBase,
+  },
+  computed: {
+    // N.b. mapGetters does not work here because of dynamic module loading (this.$data.fileType)
+    loading() {
+      return store.getters[`results/${this.fileType}/loading`];
+    },
+    error() {
+      return store.getters[`results/${this.fileType}/error`];
+    },
+    resultsTotal() {
+      return store.getters[`results/${this.fileType}/resultsTotal`];
+    },
+    pageResults() {
+      return store.getters[`results/${this.fileType}/pageResults`];
+    },
+    results() {
+      return this.pageResults(Number(this.$route.query.page) - 1 || 0);
+    },
+    shownHits() {
+      if (this.$route.query.type === this.$data.fileType) {
+        return this.results;
+      }
+      return this.results.slice(0, this.shortList);
+    },
+    pageCount() {
+      return Math.ceil(this.resultsTotal / pageSize);
+    },
+    page: {
+      get() { return Number(this.$route.query.page); },
+      set(value) {
+        this.$router.push({
+          ...this.$route,
+          query: {
+            ...this.$route.query,
+            page: value,
+          },
+        });
+      },
+    },
+  },
```

Comment:
Are these pages related to the pages from the API, are they in the store or merely in the frontend?

Reply:
That is, could you please explain (perhaps as comment in the code) how various pages relate?

Reply:
ok changed the name so it is more descriptive. Also used a generic function for updating the query.